examples/property_prediction/pretrain_gnns/chem/pretrain_masking.py

```diff
@@ -0,0 +1,214 @@
+import tqdm
+from functools import partial
+import itertools
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+
+import pandas as pd
+from random import sample
+
+import dgl
+from rdkit import Chem
+from dgllife.utils import PretrainAtomFeaturizer
+from dgllife.utils import PretrainBondFeaturizer
+from dgllife.utils import smiles_to_bigraph
+
+from model import GINMaskingModel
+from utils import *
+
+
+def compute_accuracy(pred, target):
+    return float(torch.sum(torch.max(pred.detach(), dim=1)[1] == target).cpu().item()) / len(pred)
+
+
+def collate_masking(graphs, args):
+    """
+    Collate function used for pretrain_masking task to randomly mask nodes/edges
+    """
+    bg = dgl.batch(graphs)
+    # masked nodes/edges indices in batched graph
+    masked_nodes_indices = torch.LongTensor(
+        sample(range(bg.number_of_nodes()), int(bg.number_of_nodes() * args.mask_rate)))
+    if args.mask_edge == 1:
+        masked_edges_indices = mask_edges(bg, masked_nodes_indices)
+
+    node_feats = [
+        bg.ndata.pop('atomic_number'),
+        bg.ndata.pop('chirality_type')
+    ]
+    edge_feats = [
+        bg.edata.pop('bond_type'),
+        bg.edata.pop('bond_direction_type')
+    ]
+    # export masked nodes labels
+    masked_nodes_labels = node_feats[0][masked_nodes_indices]
+    # mask these nodes labels
+    node_feats[0][masked_nodes_indices] = 118
+
+    if args.mask_edge == 1:
+        # export masked edges labels
+        masked_edges_labels = edge_feats[0][masked_edges_indices]
+        # mask these edges labels
+        edge_feats[0][masked_edges_indices] = 5
+    else:
+        # if no edge masking
+        masked_edges_indices = 0
+        masked_edges_labels = 0
+
+    # return batched graph, node features, edge features,
+    # masked node indices, masked edge indices, masked nodes labels, masked edges labels
+    return bg, node_feats, edge_feats, \
+           (masked_nodes_indices, masked_edges_indices), (masked_nodes_labels, masked_edges_labels)
+
+
+def train(args, model_list, train_dataloader, optimizer, criterion, device):
+    if args.mask_edge == 1:
+        model, node_linear, edge_linear = model_list
+    else:
+        model, node_linear = model_list
+
+    for epoch in range(1, args.epochs):
+        model.train()
+        node_linear.train()
+        if args.mask_edge == 1:
+            edge_linear.train()
+        with tqdm.tqdm(train_dataloader) as tq_train:
+            for step, (bg, node_feats, edge_feats, masked_indices, masked_labels) in enumerate(tq_train):
+                masked_nodes_indices, masked_edges_indices = masked_indices
+                masked_nodes_labels, masked_edges_labels = masked_labels
+
+                node_feats = [n.to(device) for n in node_feats]
+                edge_feats = [e.to(device) for e in edge_feats]
+                masked_nodes_indices = torch.LongTensor(masked_nodes_indices).to(device)
```

Comment:
I think you already did `masked_nodes_indices = torch.LongTensor(masked_nodes_indices)` in the collate function.

Reply:
Removed.